### PR TITLE
Fix #1353  Improve endorsement draws in api get_addresses endpoint

### LINKED ISSUE #1353: Improve endorsement draws in api get_addresses endpoint
In GitLab by @AureliaDolo

```
pub struct AddressInfo {
    // other fields
    pub endorsement_draws: HashMap<String, u64>, // u64 is the index
}
```

```
pub struct AddressInfo {
    // other fields
    pub endorsement_draws: HashSet<ESlot>, // u64 is the index
}

/// When an address is drawn to create an endorsement it is selected for a specific index
pub struct ESlot {
    slot: Slot,
    index: u64,
}
```

ESlot should be used at least in address info, but it may be useful in other places. Don't forget to update the api.md file

### PR DESCRIPTION
**Pull request by AureliaDolo**

### CODE REVIEW

**Comment by AureliaDolo** [comment]
And it looks ok in json :tada:

**Comment by AureliaDolo** [comment]
@Noxhaa @qdrn you might want to get a look at that, as it's impacting the output of get_addresses command

**Review by yvan-sraka — APPROVED** [review]

File: api/src/public.rs [hunk 1]
```diff
@@ -415,7 +415,22 @@ impl Endpoints for API<Public> {
...
                         .collect(),
-                    endorsement_draws: Default::default(), // TODO: update wait for !238
+                    endorsement_draws: next_draws
+                        .iter()
+                        .map(|(slot, (_, addrs))| {
+                            addrs.iter().enumerate().filter_map(|(index, ad)| {
+                                if *ad == address {
+                                    Some(IndexedSlot {
+                                        slot: *slot,
+                                        index: index,
```
[the remark this answers is not in the document]
Comment (AureliaDolo) [reply api/src/public.rs:425]: Yes it's the index of the endorsement that may not be the index in the vec of endorsements

**Review by yvan-sraka — APPROVED** [review]

**Review by adrien-zinger — APPROVED** [review]

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors cancel

File: api/src/public.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AureliaDolo) [reply api/src/public.rs]: To put it in a nutshell, when you are drawn for an endorsement you are drawn for a slot and at an index for that slot. But when endorsement are included in blocks, there may be missing endorsements, so the indexes in the endorsement vec may not be aligned with the indexes that one was drawn for. That's why there is a index field in the endorsement content struct.

[the remark this answers is not in the document]
Comment (adrien-zinger) [reply api/src/public.rs]: Ok nice explanation :) thx!

**Comment by yvan-sraka** [comment]
bors merge

**Pull request merged**